src/common/LinkParser.cpp

```diff
@@ -121,13 +145,14 @@ LinkParser::LinkParser(const QString &unparsedString)
 {
     ParsedLink result;
     // This is not implemented with a regex to increase performance.
-    QStringView remaining(unparsedString);
+    QString sanitizedString = sanitizeUrl(unparsedString);
```

Comment:
This must be stored in `result.source`, because all string views reference this.

Reply:
I'm not sure what you mean here, `unparsedString` goes into source, just like it did before. Should I put the sanitized version in it?

Reply:
`host` and `rest` are `QStringView`s (i.e. pointers) into `sanitizedString` - not `unparsedString`. If that string isn't kept alive after the constructor is done, both views will point to freed memory. You could alternatively add a field to the `ParsedLink`.